evm/src/executor/backend/mod.rs

```diff
@@ -1665,22 +1680,34 @@ fn is_contract_in_state(journaled_state: &JournaledState, acc: Address) -> bool
 }
 
 /// Executes the given transaction and commits state changes to the database _and_ the journaled
-/// state
+/// state, with an optional inspector
 fn commit_transaction(
     tx: Transaction,
     mut env: Env,
     journaled_state: &mut JournaledState,
     fork: &mut Fork,
+    fork_id: &ForkId,
+    cheatcodes_inspector: Option<&mut Cheatcodes>,
 ) {
     configure_tx_env(&mut env, &tx);
-    let (_, state) = {
+
+    let (result, state) = {
         let mut evm = EVM::new();
         evm.env = env;
-        evm.database(&mut fork.db);
-        evm.transact()
+
+        let db = Backend::new_with_fork(fork_id, fork.clone(), journaled_state.clone());
+        evm.database(db);
```

Comment:
this is no longer required, can just use cheatcodes inspector?

Reply:
I think it's still required because `Cheatcodes` only implements `Inspector<DB: DatabaseExt>` and so we need to provide a `DatabaseExt` to the `EVM` and the only type that implements `DatabaseExt` as of now is `Backend` (`ForkDB` does not)

Reply:
right.